allen-cell-animated/simularium-website
# Fix/update lint-staged from 8.1.4 to 10.5.4

Problem
=======
Upon `npm install` this repo was showing `10 vulnerabilities (3 low, 4 moderate, 3 high)`. The 3 high vulnerabilities were caused by:
```
property-expr  <2.0.3
Severity: high
Prototype Pollution in property-expr - https://npmjs.com/advisories/1681
fix available via `npm audit fix --force`
Will install lint-staged@10.5.4, which is a breaking change
node_modules/property-expr
  yup  <=0.28.1
  Depends on vulnerable versions of property-expr
  node_modules/yup
    lint-staged  8.1.1 - 8.2.1
    Depends on vulnerable versions of yup
    node_modules/lint-staged
```

Solution
========
I upgraded lint-staged from v8.1.4 to v10.5.4.

That removed the 3 high vulnerabilities but required updating .lintstagedrc:

* Changed the format - see v9 breaking change [here](https://github.com/okonet/lint-staged/releases/tag/v9.0.0) "The advanced configuration options have been deprecated in favour of the simple format"
* Removed `"git add"` - see v10 breaking change [here](https://github.com/okonet/lint-staged/releases/tag/v10.0.0) "Please remove `git add` from your configuration after upgrading."

## Type of change
Please delete options that are not relevant.


## Steps to verify
1. Pull this branch and run `npm i`, there should be no high severity vulnerabilities anymore.
1. Make a test commit that has a poorly formatted line of code (and also change something else in the file so that it won't reject an empty commit)
2. After committing see that the formatter did its job
